helper/resource/testing_new_config.go

```diff
@@ -443,5 +451,19 @@ func testStepNewConfig(ctx context.Context, t testing.T, c TestCase, wd *plugint
 		}
 	}
 
+	if refreshAfterApply && !step.Destroy && !step.PlanOnly {
+		if len(c.Steps) > stepIndex+1 {
+			// If the next step is a refresh, then we have no need to refresh here
+			if !c.Steps[stepIndex+1].RefreshState {
+				// Echo a searchable message to easily determine when this is no longer being used
+				fmt.Println(EnvTfAccRefreshAfterApply+":", "running apply -refresh-only -refresh=true")
```

Comment:
nit: We should probably use the `logging` package for this with whatever log level you think is appropriate